docs/domains/Domain.json

```diff
@@ -183,6 +183,14 @@
           "description": "Model in image model files and properties.",
           "$ref": "#/definitions/Model"
         },
+        "distributionStrategy": {
+          "description": "The strategy for handling changes to the secrets or overridesConfigMap fields. Legal values are DYNAMIC and ON_RESTART.",
```

Comment:
or overridesConfigMaps fields. --> or overridesConfigMaps fields once the introspectVersion is changed.  

- Or something similar. 
- The idea is to make it clear that the strategy is related to handling a change to the introspectVersion field.  
- Do the same edit for all related discussion of distributionStrategy since they're all tied to introspectVersion.
- Also, consider changing the name to 'introspectVersionDistributionStrategy' or similar...

Reply:
As far as I know, this has nothing to do with introspection. It is triggered by a change to the specified fields in the domain resource only.

Reply:
Wait, what?  We need to discuss with @rjeberhard

Reply:
Here is my understanding of this field. In operator 2.5.0, introspection could only happen on initial domain creation or following a cold shutdown, so no servers presently running. Therefore, we had no issues about how changes to any of the inputs that affect the final configuration overrides would affect running servers. Now, we have a specific need to run introspection when servers are running and even to distribute updated configuration overrides to those running servers. However, since we can't test every combination of changes to these various inputs we introduced two guards: 1) The introspectVersion, which lets the customer explicitly trigger introspection and thereby confirming that all of the inputs are ready, and 2) the distribution strategy, which lets the customer select not distributing the updated strategy dynamically to already running servers.

Reply:
I guess I don't know what Russ means by 'It' and 'specified fields' in 'It is triggered by a change to the specified fields in the domain resource only.'  

I've been assuming a re-introspect for a running in-pv or in-image domain only occurs if the introspectVersion changes.  Is that wrong?

Reply:
Or following a cold restart, as before. We had to increase the cases where we ran introspection for model-in-image, but that was not supposed to affect in-pv or image.

Reply:
I know what he might mean. There was a question about how and when changes to the distribution strategy field itself should be activated. Since we proposed that the distribution strategy would be realized as an env. variable in the server container then the active distribution strategy would be the value set when a given pod was started.

Reply:
So, back to my original comment in this thread, is it valid?

Reply:
We tell customers about introspection, so we can reference that.  How about... "The strategy for how configuration overrides are distributed to WebLogic servers following introspection. Configuration overrides are generated during introspection from secrets, the overrideConfigMap field, and WebLogic domain topology. Legal values are DYNAMIC and ON_RESTART."

Reply:
Ah, so simply changing the domain spec to reference a different override config map will not  have any effect until the next introspection?

Reply:
That's correct.  It's also intentional here since the customer will likely make coordinating changes to secrets, the input overrides config map, and even domain topology (using online WLST) and so we only want to do introspection when the customer is ready.

Reply:
What about changing the distribution strategy? Should that force the pods to roll? Currently it does, which seems troubling. 

I've tried to be specific about the effects of this feature by writing up a testing JIRA https://jira.oraclecorp.com/jira/browse/OWLS-82325. I think that may be the best way to communicate about it, and based on what you've just said, it's wrong.

Reply:
Yes, I agree that changing the distribution strategy should not cause the pods to roll. I'd prefer a behavior like, "the active distribution strategy is the distribution strategy configured when the server was started."  We have the same behavior for the shutdown-related options.

Reply:
Can you point to code for those shutdown-related options?

Reply:
"So it sounds to me as though what we're now calling the domain config map is the one which contains the generated overrides, as opposed to the overrideConfigMap, which contains information for describing the overrides. "   Again, calling it 'domain config map' is quite confusing name IMO, for the reasons I stated in the other thread.    How about 'introspector config map, which contains output from the introspector, as opposed to the override and model config maps, which are inputs to the introspector'.

Reply:
Ryan:  How about  "The strategy for how configuration overrides are distributed to WebLogic servers following introspection. Configuration overrides are generated during introspection from secrets, the overrideConfigMap field, and WebLogic domain topology. Legal values are DYNAMIC and ON_RESTART."

Sounds good to me.  Please also add:   "The default is WHATEVER."   And  please also use the same wording in the other locations that describe the new field.

Reply:
The strategy for configuration override distribution to WebLogic servers following introspection. Configuration overrides are generated during introspection from secrets, the overrideConfigMap field, and WebLogic domain topology. Legal values are DYNAMIC and ON_RESTART. The default value is XXXXX. Changes to the strategy require a full domain restart to take effect.